pkg/detector/preemption.go

```diff
@@ -227,3 +228,41 @@ func (d *ResourceDetector) fetchResourceTemplate(rs policyv1alpha1.ResourceSelec
 	}
 	return resourceTemplate, nil
 }
+
+// HandleDeprioritizedPropagationPolicy responses to priority change of a PropagationPolicy,
+// if the change is from high priority (e.g. 5) to low priority(e.g. 3), it will
+// check if there is another PropagationPolicy could preempt the targeted resource,
+// and put the PropagationPolicy in the queue to trigger preemption.
+func (d *ResourceDetector) HandleDeprioritizedPropagationPolicy(oldPolicy policyv1alpha1.PropagationPolicy, newPolicy policyv1alpha1.PropagationPolicy) {
+	if (newPolicy.Spec.Priority != nil && oldPolicy.Spec.Priority != nil) && *newPolicy.Spec.Priority < *oldPolicy.Spec.Priority {
+		klog.Infof("PropagationPolicy(%s/%s) priority changed from %d to %d", newPolicy.GetNamespace(), newPolicy.GetName(), *oldPolicy.Spec.Priority, *newPolicy.Spec.Priority)
+		policies, err := d.propagationPolicyLister.ByNamespace(newPolicy.GetNamespace()).List(labels.Everything())
+		if err != nil {
+			klog.Errorf("Failed list PropagationPolicy from namespace: %s, error: %v", newPolicy.GetNamespace(), err)
+			return
+		}
+
+		// TODO(@RainbowMango): Should sort the listed policies to ensure the
+		// higher priority PropagationPolicy be process first to avoid possible
+		// multiple preemption.
+
+		for i := range policies {
+			var potentialPolicy policyv1alpha1.PropagationPolicy
+			if err = helper.ConvertToTypedObject(policies[i], &potentialPolicy); err != nil {
+				klog.Errorf("Failed to convert typed PropagationPolicy: %v", err)
+				continue
+			}
+			if potentialPolicy.Spec.Priority != nil &&
+				potentialPolicy.Spec.Preemption == policyv1alpha1.PreemptAlways &&
+				*potentialPolicy.Spec.Priority > *newPolicy.Spec.Priority { // potential policy must have higher priority.
```

Comment:
I think potentialPolicy's priority must be <= oldPolicy.Spec.Priority. Otherwise, there have been occured the preemption.

Reply:
Updated.
We can save the effort for this case.